File: 351004/Nepesov/Lab1/untitled/src/main/java/org/example/mapper/NewsMapper.java
package org.example.mapper;

import org.example.dto.NewsRequestTo;
import org.example.dto.NewsResponseTo;
import org.example.model.News;
import org.mapstruct.Mapper;
import org.mapstruct.MappingConstants;

@Mapper(componentModel = MappingConstants.ComponentModel.SPRING)
public interface NewsMapper {
    News toEntity(NewsRequestTo requestTo);
    NewsResponseTo toResponse(News entity);
}

File: 351003/brezgunov/distcomp/src/main/kotlin/com/distcomp/mapper/NewsMapper.kt
package com.distcomp.mapper

import com.distcomp.dto.news.NewsRequestTo
import com.distcomp.dto.news.NewsResponseTo
import com.distcomp.entity.News
import org.mapstruct.Mapper
import org.mapstruct.MappingConstants
import org.mapstruct.ReportingPolicy

@Mapper(
    componentModel = MappingConstants.ComponentModel.SPRING,
    unmappedTargetPolicy = ReportingPolicy.IGNORE
)
interface NewsMapper {
    fun toNewsResponse(news: News) : NewsResponseTo

    fun toNewsEntity(newsRequestTo: NewsRequestTo) : News
}